### PR TITLE
cpu/stm32f1: reworked timer implementation

### PR DESCRIPTION
**Pull request by haukepetersen**
Cleaned up the timer driver for the stm32f1. The old implementation concatenated two 16-bit timers to force a 32-bit timer. This did not bring many benefits and is not needed anymore, as the xtimer works fine with 16-bit timers. Also this complicated the code quite a bit while still throwing interrupts on every 16-bit timer overflow...

```
tests/periph_timer @ master:
   text	   data	    bss	    dec	    hex	
   9040	    128	   2808	  11976	   2ec8

test/periph_timer @ this PR:
   text	   data	    bss	    dec	    hex	
   8556	    128	   2808	  11492	   2ce4
```
savings in ROM: almost 500byte!

### CODE REVIEW

File: cpu/stm32f1/include/periph_cpu.h [hunk 1]
```diff
@@ -44,6 +44,16 @@ typedef uint32_t gpio_t;
 #define GPIO_PIN(x, y)      ((GPIOA_BASE + (x << 10)) | y)
 
 /**
+ * @brief   All timers for the STM32F0 have 4 CC channels
```
Comment (thomaseichinger) [inline cpu/stm32f1/include/periph_cpu.h]: s/F0/F1/

**Comment by thomaseichinger** [comment]
@haukepetersen ACK when the minor comment is fixed and Travis passed one day.

**Comment by haukepetersen** [comment]
fixed and squashed.

**Comment by PeterKietzmann** [comment]
BTW: I ran several experiments on an iotlab-m3 node with this PR as a basis. Worked without problems.

**Comment by PeterKietzmann** [comment]
and go

**Pull request merged**